packages/cli/src/commands/doctor/doctor.js

```diff
@@ -0,0 +1,131 @@
+import chalk from 'chalk';
+import envinfo from 'envinfo';
+import {logger} from '@react-native-community/cli-tools';
+import ISSUES from './issues';
+import {getLoader} from '../../tools/loader';
+import printFixOptions, {KEYS} from './printFixOptions';
+import runAutomaticFix, {AUTOMATIC_FIX_LEVELS} from './runAutomaticFix';
+
+const printCategory = ({label, key}) => {
+  if (key > 0) {
+    logger.log();
+  }
+
+  logger.log(chalk.dim(label));
+};
+
+const printIssue = ({label, needsToBeFixed, isRequired}) => {
+  const symbol = needsToBeFixed
+    ? isRequired
+      ? chalk.red('✖')
+      : chalk.yellow('●')
+    : chalk.green('✓');
+
+  logger.log(` ${symbol} ${label}`);
+};
+
+export default (async function runDoctor() {
+  const Loader = getLoader();
+  const loader = new Loader();
```

Comment:
Nit: naming this a "loader" when it's not really showing a loading message and is mostly only a logger is hard to read (I recognize that this is called a loader elsewhere so we should either change all usages or stay consistent)